SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -1068,28 +1068,50 @@ void USpatialNetDriver::NotifyActorDestroyed(AActor* ThisActor, bool IsSeamlessT
 		// Check if this is a dormant entity, and if so retire the entity
 		if (PackageMap != nullptr && World != nullptr)
 		{
-			const Worker_EntityId EntityId = PackageMap->GetEntityIdFromObject(ThisActor);
-
-			// If the actor is an initially dormant startup actor that has not been replicated.
-			if (EntityId == SpatialConstants::INVALID_ENTITY_ID && ThisActor->IsNetStartupActor() && ThisActor->GetIsReplicated()
-				&& ThisActor->HasAuthority())
+			if (!World->bBegunPlay)
 			{
-				UE_LOG(LogSpatialOSNetDriver, Log,
-					   TEXT("Creating a tombstone entity for initially dormant statup actor. "
-							"Actor: %s."),
-					   *ThisActor->GetName());
-				ActorSystem->CreateTombstoneEntity(ThisActor);
+				// When running in PIE, Blueprint loaded levels can be duplicated and immediately unloaded.
+				// This causes each actor in the level to be destroyed, which we catch here and attempt to create a tombstone for.
+				// This is incorrect behaviour, as the level was never actually part of the scene, just an artifact from the
+				// world duplication process. So ignore this destroy call if the world hasn't begun play, and it was duplicated from PIE.
+				if (ThisActor->GetLevel() != nullptr && ThisActor->GetLevel()->bWasDuplicatedForPIE)
+				{
+					UE_LOG(LogSpatialOSNetDriver, Verbose,
```

Comment:
nit - might be useful for these two logs to be different?

Reply:
nit - if we don't need different messages, if there any need for the if statement since we are going to ignore anyway?

Reply:
Could alter the message to be slightly different in the expectant case. The reason I added it was because the behaviour is still somewhat unusual, and wanted to give us the chance to more easily debug in future if necessary.